src/reducers/MapReducer.js

```diff
@@ -25,7 +25,7 @@ export default (state = INITIAL_STATE, action) => {
     switch (type) {
 
         case INITIALIZE_MAP:
-            return INITIAL_STATE;
+            return payload ? {...INITIAL_STATE, region: payload} : INITIAL_STATE;
```

Comment:
When the `INITIALIZE_MAP` action comes in, if there's a payload (which is the region in this case) from rehydration, use that. If there isn't, then pass along the `INITIAL_STATE` for the map.